### PR TITLE
fix queue problem

### PR DESCRIPTION
**Pull request by shalehaha**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by shalehaha** [comment]
@nporter-adbe @cdhoffmann @sbenedicadb 
Let me know if there is a better way to fix this.

File: AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift [hunk 1]
```diff
@@ -63,7 +63,9 @@ public class PersistentHitQueue: HitQueuing {
             guard !self.suspended else { return }
             guard let hit = self.dataQueue.peek() else { return } // nothing left in the queue, stop processing
 
-            self.processor.processHit(entity: hit, completion: { [weak self] success in
+            
+            let semaphore = DispatchSemaphore(value: 0)
```
Comment (shalehaha) [inline AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift:67]: need to block the current dispatchqueue until getting the respsonse

Comment (nporter-adbe) [reply AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift:67]: Yeah, I can see how the bug could occur... I think this is probably the best solution, potentially using an `OperationQueue` from the Swift library could help us here but I don't think that's required.

**Review by nporter-adbe — APPROVED** [review]

**Comment by nporter-adbe** [comment]
The only way I can see this bug happening is if `beginProcessing` is called while we are already processing hits. Another fix might to be only call `processNextHit` if we're not already suspended.

**Comment by shalehaha** [comment]
@nporter-adbe I thought about using `suspended`, but it could mess things up with privacy status change. Not sure if introducing another bool flag similar to `suspended` can solve this problem.

**Pull request merged**